Jenkinsfile_updatecli

```diff
@@ -1,68 +1,43 @@
 def cronExpr = env.BRANCH_IS_PRIMARY ? 'H/30 * * * *' : ''
 
-pipeline {
-  agent {
-    label 'jnlp-linux-arm64'
-  }
-
-  options {
-    buildDiscarder(logRotator(numToKeepStr: '10'))
-    timeout(time: 30, unit: 'MINUTES')
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '10')),
     disableConcurrentBuilds()
-  }
+])
 
-  triggers {
-    cron (cronExpr)
-  }
+node('jnlp-linux-arm64') {
+  timeout(time: 30, unit: 'MINUTES') {
+    if (cronExpr) {
+        properties([pipelineTriggers([cron(cronExpr)])])
+    } 
+      // Set environment variable for Azure credentials (from Jenkins credentials)
+      env.UPDATECLI_AZURE = credentials('updatecli-azure-serviceprincipal')
```

Comment:
This directive is only for declarative pipeline. It is translated into a `withCredential()` step under the hood: You have to use the `withCredentials()` below and append the required credential, otherwise the pipeline fails with `az login --service-principal -u '' -p '' -t ''` (empty variable)